lib/daru/dataframe.rb

```diff
@@ -295,11 +295,11 @@ def initialize source, opts={}
           end
         when Hash
           create_vectors_index_with vectors, source
+          vectors_have_same_index = all_vectors_have_equal_indexes?(source)
```

Comment:
`vectors_have_same_index` is referenced in the `if clone` block.  When a dataframe is duplicated, all vectors have the same index, but `vectors_have_same_index` is false because `!index.nil?` will be true.  So when a dataframe is duplicated, all of the vectors have to be recreated, rather than just using the `.dup` method on the vector (line 321 below).

I don't think I fully follow your second point.  Could you provide a test that would fail and I'll work to get it passing?